src/test/java/edu/cornell/kfs/module/purap/service/impl/JaggaerRoleServiceImplTest.java

```diff
@@ -0,0 +1,235 @@
+package edu.cornell.kfs.module.purap.service.impl;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.config.Configurator;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.kuali.kfs.kim.api.identity.Person;
+import org.kuali.kfs.kim.api.permission.PermissionService;
+import org.kuali.kfs.sys.KFSConstants;
+import org.mockito.Mockito;
+
+import edu.cornell.kfs.module.purap.CUPurapConstants;
+import edu.cornell.kfs.module.purap.CUPurapConstants.JaggaerRoleSet;
+import edu.cornell.kfs.module.purap.JaggaerConstants;
+import edu.cornell.kfs.module.purap.businessobject.JaggaerRoleLinkMapping;
+import edu.cornell.kfs.module.purap.service.JaggaerRoleLinkMappingService;
+
+class JaggaerRoleServiceImplTest {
+    
+    private JaggaerRoleServiceImpl jaggaerRoleService;
+    private Person user;
+    
+    private static final String JAGGAER_ROLE_HAS = "jaggar role has";
+    private static final String JAGGAER_ROLE_DOES_NOT_HAVE = "jaggar role does not have";
```

Comment:
Minor typos in string values of these test roles.

Reply:
ok